pds4indextools/pds4_create_xml_index.py

```diff
@@ -1,69 +1,200 @@
-    --extra-file-info EXTRA_FILE_INFO  
+    --add-extra-file-info ADD_EXTRA_FILE_INFO
                          Add additional column(s) to the index file containing file or
                          bundle information. Possible values are: "LID", "filename",
                          "filepath", "bundle", and "bundle_lid". Multiple values may be
                          specified separated by spaces.
     --config-file CONFIG_FILE
                          An optional .ini configuration file for further customization.
+    --dump-available-xpaths
+                         Create a .txt file containing all available XPath headers for
+                         given label file(s). Can be modified and used as a file for
+                         --elements-file
+    --fixed-width        Create a fixed-width index file.
+    --generate-label     Create a label file (Product_Ancillary or
+                         Product_Metadata_Supplemental) for a generated index
+                         file.
+    --label-user-input   Add additional content to the generated label file.
 
 Example:
-python3 pds4_create_xml_index.py <toplevel_directory> "glob_path1" "glob_path2" 
---output_file <outputfile> --elements-file sample_elements.txt --verbose
+    python3 pds4_create_xml_index.py <toplevel_directory> "glob_path1" "glob_path2"
+    --output_file <outputfile> --elements-file sample_elements.txt --verbose
 """
 
 import argparse
+from collections import namedtuple
+import csv
 import configparser
+from datetime import datetime
+import fnmatch
+import functools
+import itertools
+from itertools import groupby
 from lxml import etree
+import os
 import pandas as pd
 from pathlib import Path
+import platform
 import requests
 import sys
+import yaml
+
+import pdstemplate as ps
+
+
+SplitXPath = namedtuple('SplitXPath',
+                        ['xpath', 'parent', 'child', 'prefix', 'num'])
+
+
+def convert_header_to_xpath(root, xml_header_path, namespaces):
+    """
+    Replace hierarchical components of XPath with attribute names and namespaces.
+
+    While the XPaths are accurate to the hierarchy of the elements referenced, they
+    provide no information on their own without the attributed label file for reference.
+    This function replaces the asterisks with the respective names of the elements and
+    attributes they represent.
+
+    Args:
+        root (Element): The root element of the XML document.
+        xml_header_path (str): Original XML header path.
+        namespaces (dict): Dictionary of XML namespace mappings.
+
+    Returns:
+        str: Converted XPath expression.
+
+    Example:
+        >>> tree = etree.parse(str(path_to_label_file))
+        >>> root = tree.getroot()
+        >>> xml_header_path = '/*/*[1]/*[2]'
+        >>> namespaces = root.nsmap
+        >>> convert_header_to_xpath(root, xml_header_path, namespaces)
+        'pds:Product_Observational/pds:Identification_Area[1]/pds:version_id[2]'
+
```

Comment:
Put the blank line after the triple-quote, not before. Or don't have a blank line at all, which seems to be your style elsewhere.

Reply:
Huh, `flake8` didn't catch this one.